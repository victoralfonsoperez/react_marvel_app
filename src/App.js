import React, { Component } from 'react'
import { getComics } from './utils/api'
import logo from './logo.svg'
import './App.css'

class App extends Component {
  state = {}

  componentDidMount() {
    getComics().then(response => this.updateComics(response.data))
  }

  updateComics = (data) => {
    this.setState({ comics: data.results })
  }

  render() {
    return (
      <div className="App">
        {
          this.state.comics && this.state.comics.map(comic => (
            <div key={comic.id}>
              {comic.title}
            </div>
          ))
        }
      </div>
    )
  }
}

export default App
