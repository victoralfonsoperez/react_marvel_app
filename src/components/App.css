.App {
  text-align: center;
  padding-top: 60px;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.comics-container {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
