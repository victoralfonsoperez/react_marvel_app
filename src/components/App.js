import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import PropTypes from 'prop-types'
import { FaSpinner } from 'react-icons/lib/fa'
import { getCharacters } from '../utils/api'
import * as actions from '../actions'
import Header from './Header'
import './App.css'
import charactersLogo from '../assets/characters.png'
import favouritesLogo from '../assets/favourites.png'
import ComicCard from './ComicCard'
import Footer from './Footer'
import SectionHeading from './SectionHeading'

class App extends Component {
  state = {}

  componentDidMount() {
    getCharacters()
      .then(characters => this.props.getAllCharacters(characters, true))
  }

  render() {
    const { characters, ready } = this.props
    return (
      <div className="App">
        <Header />
        <section>
          <article className="comics-container">
            <SectionHeading imageSrc={charactersLogo} title="Characters" />
            {
              ready === false && (
                <FaSpinner />
              )
            }
            {
          ready && characters && characters.map(comic => (
            <ComicCard key={comic.id} comic={comic} />
          ))
        }
            {
          ready && characters && characters.length === 0 && (
            <div className="no-results">
              There are no results for this search
            </div>
          )
        }
          </article>
          <aside className="aside-container">
            <SectionHeading imageSrc={favouritesLogo} title="My favourites" />
          </aside>
        </section>
        <Footer />
      </div>
    )
  }
}

App.defaultProps = {
  characters: [],
}

App.propTypes = {
  getAllCharacters: PropTypes.func.isRequired,
  characters: PropTypes.array,
  ready: PropTypes.bool.isRequired,
}

const mapStateToProps = ({ characters, newCharacters, ready }) => (
  {
    characters: characters && characters.data && characters.data.results,
    newCharacters: newCharacters && newCharacters.data && newCharacters.data.results,
    ready,
  }
)

const mapDispatchToProps = dispatch => (
  bindActionCreators(actions, dispatch)
)

export default connect(mapStateToProps, mapDispatchToProps)(App)
