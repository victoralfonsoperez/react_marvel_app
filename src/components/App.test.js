import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import { App } from './App'

describe('App', () => {
  const mockGetCharacters = jest.fn()
  const props = { getAllCharacters: mockGetCharacters, ready: false }
  const app = shallow(<App {...props} />)

  it('renders properly', () => {
    expect(app).toMatchSnapshot()
  })
})
