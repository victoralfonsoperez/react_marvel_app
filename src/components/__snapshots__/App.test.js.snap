// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    characters={Array []}
    getAllCharacters={[Function]}
    ready={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <section>
          <article
                    className="comics-container"
          >
                    <SectionHeading
                              imageSrc="characters.png"
                              title="Characters"
                    />
                    <FaSpinner />
          </article>
          <aside
                    className="aside-container"
          >
                    <SectionHeading
                              imageSrc="favourites.png"
                              title="My favourites"
                    />
          </aside>
</section>,
        <Connect(Footer) />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <article
              className="comics-container"
>
              <SectionHeading
                            imageSrc="characters.png"
                            title="Characters"
              />
              <FaSpinner />
</article>,
            <aside
              className="aside-container"
>
              <SectionHeading
                            imageSrc="favourites.png"
                            title="My favourites"
              />
</aside>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <SectionHeading
                  imageSrc="characters.png"
                  title="Characters"
/>,
                <FaSpinner />,
                false,
                false,
              ],
              "className": "comics-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "imageSrc": "characters.png",
                  "title": "Characters",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              false,
              false,
            ],
            "type": "article",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <SectionHeading
                imageSrc="favourites.png"
                title="My favourites"
/>,
              "className": "aside-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "imageSrc": "favourites.png",
                "title": "My favourites",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "aside",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <section>
            <article
                        className="comics-container"
            >
                        <SectionHeading
                                    imageSrc="characters.png"
                                    title="Characters"
                        />
                        <FaSpinner />
            </article>
            <aside
                        className="aside-container"
            >
                        <SectionHeading
                                    imageSrc="favourites.png"
                                    title="My favourites"
                        />
            </aside>
</section>,
          <Connect(Footer) />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <article
                className="comics-container"
>
                <SectionHeading
                                imageSrc="characters.png"
                                title="Characters"
                />
                <FaSpinner />
</article>,
              <aside
                className="aside-container"
>
                <SectionHeading
                                imageSrc="favourites.png"
                                title="My favourites"
                />
</aside>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <SectionHeading
                    imageSrc="characters.png"
                    title="Characters"
/>,
                  <FaSpinner />,
                  false,
                  false,
                ],
                "className": "comics-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "imageSrc": "characters.png",
                    "title": "Characters",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                false,
                false,
              ],
              "type": "article",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <SectionHeading
                  imageSrc="favourites.png"
                  title="My favourites"
/>,
                "className": "aside-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "imageSrc": "favourites.png",
                  "title": "My favourites",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "aside",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
